vulnerable_people_form/form_pages/shared/routing.py

```diff
@@ -158,10 +160,11 @@ def _get_next_form_url_based_on_location_tier(redirect_url, should_redirect_to_n
 def _get_next_form_url_after_shopping_and_priority_supermarket():
     if _is_tiering_logic_enabled():
         location_tier = get_location_tier()
+        shielding_advice = get_shielding_advice()
```

Comment:
This method needs updating `get_redirect_for_returning_user_based_on_tier` as it currently doesn't check the shielding advice - this is for an nhs login user returning back to their submission as well as some one on check answers clicking the change address